sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -127,33 +127,41 @@ abstract class LogicalPlan extends QueryPlan[LogicalPlan] with Logging {
       cleanLeft.children.size == cleanRight.children.size && {
       logDebug(
         s"[${cleanRight.cleanArgs.mkString(", ")}] == [${cleanLeft.cleanArgs.mkString(", ")}]")
-      cleanRight.cleanArgs == cleanLeft.cleanArgs
```

Comment:
how about we just change this to:
```
cleanRight.zip(cleanArgs).forall {
  case (e1: Expression, e2: Expression) => e1 semanticEquals e2 
  caes (a1, a2) => a1 == a2
}
```
then we can just improve `Expression.sentaicEquals`